terraform/modules/ecr/main.tf

```diff
@@ -20,6 +20,17 @@ resource "aws_ecr_repository" "this" {
   }
 }
 
+#tfsec:ignore:aws-ecr-repository-customer-key:encryption using KMS CMK not required
+resource "aws_ecr_repository" "db_migration" {
+  #checkov:skip=CKV_AWS_136:encryption using KMS not required
+  name                 = "db-migration"
+  image_tag_mutability = "IMMUTABLE"

```

Comment:
I think it maybe preferable in this case to not have IMMUTABLE images and let them have the same "latest" tag? Any thoughts on this? 

I'm also wondering if we should update our pipelines to automatically build the DB migration image and push it to the ECR, but I feel this is probably unnecessary given we're expecting to only use it a handful of times max.

Reply:
Why not have immutable images? For convenience? If we do have immutable images, we'll need to think of a sensible tagging strategy (I'm assuming we won't want commit refs, as the contents of the image will be determined by the contents of the db rather than the state of the code at the time of running the Dockerfile). But we could definitely use timestamps to easily generate unique names.

Reply:
Yeah more for convenience. At the moment we just tag the image with `latest` and so anytime we want to put another migration image in the ECR, we have to do delete the old `latest` first due to immutability. And in this case I'm not sure we need multiple images with different tags. We're also building and uploading the images locally, so any tagging strategy needs to be easy to do here.

Reply:
Not sure why my comment appeared after Rachael's when I wrote it a while before 😄, nonetheless I'll go with mutable and remove the ignore changes in the terraform 👍